### ISSUE TITLE
Weird character in .abapgit.xml

### ISSUE THREAD

**Issue by flaiker** [larshp/abapGit#629]
I just edited a .abapgit.xml file in GitHub's web interface and noticed this character:
![image](https://cloud.githubusercontent.com/assets/5270359/23065197/9cd2ca56-f514-11e6-87f2-b18232b13d59.png)

If I delete it in the webinterface it will be recreated in abapGit (shown as diff).
![image](https://cloud.githubusercontent.com/assets/5270359/23065217/bcee23e4-f514-11e6-9d59-e55baa2a1d7f.png)

Not sure why it's there of where it comes from but I don't think that was the case in previous abapGit versions.

**Comment by flaiker** [comment]
Not sure if related, but if I try to commit the recreated .abapgit.xml file in abapGit I get this:
![image](https://cloud.githubusercontent.com/assets/5270359/23065471/01785268-f516-11e6-880a-ce423b346fd1.png)

**Comment by sbcgua** [comment]
JS error: this is not related to the .abapgit. Does the Advanced->Debug info page work for you ? In particular do you see grey `js: OK` the the bottom right corner ? Could you also tell if the Red message stays or dissapreares after a small delay? Also if Stage page displays anything ?

**Comment by larshp** [comment]
there is a workaround here https://github.com/larshp/abapGit/blob/4ca551b74e6816fedfcf99438ddc6c97814ed532/src/zabapgit_html_action_utils.prog.abap#L272 for a similar issue when editing the xml database content via Advanced -> Database util

I have no idea why sometimes the character is created, but it seems only to happen on some SAP versions/systems

**Comment by sbcgua** [comment]
@larshp it looks like a unicode byte order prefix. I think one option is to always check its presence in XML and remove if it is there.

`CL_ABAP_CHAR_UTILITIES=>BYTE_ORDER_MARK_BIG = 'FEFF'` (why BIG endian it is mistery though indeed ...)

**Comment by larshp** [comment]
yeah

@flaiker can you check if the problem is only for .abapgit.xml or all .xml files in the repository?

**Comment by flaiker** [comment]
Hmm. So I think the stage page does not work (JS Init Error) because there are no other files that have changed. If I change some other code it will show up in the stage page but abapgit.xml does not show up there.
If I remove the char on abapgit.xml on GitHub and try to pull the change in abapGit it will show the whole file as a diff and not pull it. Only if I change another line on GitHub I can pull that change. The first char is still marked as a change that cannot be pulled ("remote: modified").
I added a test.xml to the repository to see if something happens with it but I cannot seem to pull it in abapGit. Looking at other repositories I think non code files that are not abapgit.xml generally cannot be pulled either way. It remains as "Remote: added new".
Not sure if that helps.

**Comment by larshp** [comment]
can you try adding a domain or data element in SAP and pushing it to the remote repository? That way we can check if it is only the .abapgit.xml that has the problem or it is all .xml files generated by abapGit

**Comment by flaiker** [comment]
Ah you mean those xml files. They also have it.
![image](https://cloud.githubusercontent.com/assets/5270359/23076046/1204246c-f53f-11e6-88fd-c59223f4de6a.png)

**Comment by larshp** [comment]
its easiest just to do a workaround(guess somewhere in https://github.com/larshp/abapGit/blob/master/src/zabapgit_xml.prog.abap) that skips the character, feel free to PR

endianness is always complicated 😉

**Comment by flaiker** [comment]
I investigated a bit. While having the BOM at the start of a UTF-8/16 file is optional in unicode ([source](https://en.wikipedia.org/wiki/Byte_order_mark#UTF-16)) it seems to be required by XML standards. At least the documentation of ABAP's IXML library says so ([source](http://help-legacy.sap.com/abapdocu_751/en/abenabap_ixml_lib.htm)).
![image](https://cloud.githubusercontent.com/assets/5270359/23179989/223cda84-f870-11e6-991c-39acc4976289.png)

What was surprising for me was seeing it in GitHub's editor. I checked Sublime Text and Atom (which is a bit ironic) and they both hide it from the user. I contacted GitHub support about it and apparently hiding it in view mode and displaying it in edit mode is "expected behavior".

So not abapGit's fault I guess?
